libs/damap/src/lib/components/dmp/people/people.component.ts

```diff
@@ -33,33 +33,67 @@ export class PeopleComponent implements OnInit {
 
   readonly roles: any = ContributorRole;
   readonly identifierType = IdentifierType;
-  readonly translateEnumPrefix = 'enum.contributor.role.'
+  readonly translateEnumPrefix = 'enum.contributor.role.';
 
   private searchTerms = new Subject<string>();
+  private subscriptions: Subscription[] = [];
 
   searchResult$: Observable<Contributor[]>;
   serviceConfig$: ServiceConfig[];
   serviceConfigType: ServiceConfig;
 
   constructor(
     private backendService: BackendService,
-    public dialog: MatDialog,
+    public dialog: MatDialog
   ) {}
 
   ngOnInit(): void {
     this.backendService.loadServiceConfig().subscribe(service => {
       this.serviceConfig$ = service.personSearchServiceConfigs;
-      this.serviceConfigType = this.serviceConfig$[0];
+      this.serviceConfigType = service.personSearchServiceConfigs[0];
+      this.updateSearchResult();
     });
+
     this.searchResult$ = this.searchTerms.pipe(
       debounceTime(300),
       distinctUntilChanged(),
-      switchMap((term: string) =>
-        this.backendService.getPersonSearchResult(term, this.serviceConfigType.displayText)
-      )
+      switchMap((term: string) => {
+        return this.backendService.getPersonSearchResult(
+          term,
+          this.serviceConfigType.displayText
+        );
+      })
+    );
+  }
+
+  onServiceConfigChange(serviceConfigType: ServiceConfig) {
+    this.serviceConfigType = serviceConfigType;
+    this.searchTerms.next((event.target as HTMLInputElement).value);
+    this.updateSearchResult();
+  }
+
+  private updateSearchResult(): void {
+    const searchResult$ = this.searchTerms.pipe(
+      debounceTime(300),
+      distinctUntilChanged(),
+      switchMap((term: string) => {
+        return this.backendService.getPersonSearchResult(
+          term,
+          this.serviceConfigType.displayText
+        );
+      })
+    );
+    this.subscriptions.push(
+      searchResult$.subscribe(result => {
+        this.searchResult$ = of(result);
+      })
     );
   }
```

Comment:
Will this create one subscription per call? Then we would have several open unused subscriptions :thinking: 
Can be moved to the `ngOnInit` method as it has to be called only once. By subscribing, we already get all the search input from the user and the request to the backend will be performed.